### ISSUE TITLE
Pytest warning in dynamics_test

### ISSUE THREAD

**Issue by wang-chen** [pypose/pypose#152]
## Summary

when run pytest, following warnings are generated
```
test/module/test_dynamics.py::test_dynamics_floquet
  /pypose/pypose/module/dynamics.py:198: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    self._ref_state = torch.tensor(self.state) if state is None else torch.atleast_1d(state)

test/module/test_dynamics.py::test_dynamics_floquet
  /pypose/pypose/module/dynamics.py:199: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    self._ref_input = torch.tensor(self.input) if input is None else torch.atleast_1d(input)
```

## Improvements

Should we consider changing the following lines.
https://github.com/pypose/pypose/blob/a21dfa606440f1c605a0ab166ba6cd76c664f1a6/pypose/module/dynamics.py#L198
https://github.com/pypose/pypose/blob/a21dfa606440f1c605a0ab166ba6cd76c664f1a6/pypose/module/dynamics.py#L199
to 
```
self._ref_state = self.state.clone() if state is None else torch.atleast_1d(state)
self._ref_input = self.input.clone() if input is None else torch.atleast_1d(input)
```
Do you need to retain the gradient? should we use `self.state.clone()` or `self.state.clone().detach()` or `self.state.clone().detach().require_gradient(True)`?

## Another warning

```
test/module/test_dynamics.py::test_dynamics_cartpole
 /pypose/test/module/test_dynamics.py:90: UserWarning: The use of `x.T` on tensors of dimension other than 2 to reverse their shape is deprecated and it will throw an error in a future release. Consider `x.mT` to transpose batches of matricesor `x.permute(*torch.arange(x.ndim - 1, -1, -1))` to reverse the dimensions of a tensor. (Triggered internally at  /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1659484611838/work/aten/src/ATen/native/TensorShape.cpp:2985.)
    jacob_state, jacob_input = state_all[-1, :].T, input[-1]
```
I found the shape of `state_all[-1, :]` is `torch.Size([4])`, thus `state_all[-1, :].T` doesn't make sense, should we consider changing 
https://github.com/pypose/pypose/blob/a21dfa606440f1c605a0ab166ba6cd76c664f1a6/test/module/test_dynamics.py#L90
to 
```
jacob_state, jacob_input = state_all[-1], input[-1]
```

**Comment by wang-chen** [comment]
@apandey02 can you have a look at this warning?
